content/doc/developer/handling-requests/json.adoc

```diff
@@ -0,0 +1,271 @@
+        response = j.getJSON(GET_API_URL, webClient);
+        assertEquals(response.getStatusCode(), 403);
+
+        // - same call but authenticated using withBasicApiToken() should be fine
+        response = j.getJSON(GET_API_URL, webClient.withBasicApiToken(admin));
+        assertEquals(response.getStatusCode(), 200);
+    }
+
+----
+
+== Handling POST with Jenkins
+
+This section shows how to expose an HTTP endpoint that takes a structured json Object as input, and do a response with a json structured Object.
+For this exmaple the same Object is used as input and output, but you can also use different json structure for the response.
+
+Starting from the class `JsonAPI` provided for GET example, add:
+
+[source,java]
+----
+@POST
+@WebMethod(name = "")
+public JsonHttpResponse create(@JsonBody MyJsonObject body) {
+    //Do any logic required for creation
+    //For the example purpose we just recreate json from the same object.
+    JSONObject response = JSONObject.fromObject(body);
+    return new JsonHttpResponse(response, 200);
+}
+----
+
+=== Testing with CURL
+
+A `mvn hpi:run` in the plugin should be enough to run it locally.
+Assuming that Jenkins is up at http://localhost:8080/jenkins/ , you should have at this point:
+
+Write a file `my.json` containing the JSON body:
+[source,bash]
+----
+{"message":"A nice message to send"}
+----
+
+Then if you need a user and a token:
+
+* go on Jenkins UI
+* login as a user, for example 'myuser'
+* on the top right click on user name
+* go on configure (for this user)
+* in the section "API Token" create a new token.
+
+For additional documentation on token, please visit:
+
+* link:../../../book/system-administration/authenticating-scripted-clients[Authenticating scripted clients]
+* link:../../../book/security/csrf-protection[CSRF Protection]
+
+And then send the POST request:
+[source,bash]
+----
+curl -XPOST \
+    -H "Content-Type: application/json" \
+    --user myuser:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx \
+    http://localhost:40393/jenkins/testing-cli/postSomething \
+    --data "@/my.json" \
+
+{"message":"A nice message to send"}
+ STATUS:200
+----
+
+=== Example of test with JenkinsRule
+
+Starting from the class `JsonAPITest` provided for the GET example, add:
+
+[source,java]
+----
+@Test
+public void testPostJSON() throws Exception {
+
+    MyJsonObject objectToSend = new MyJsonObject("Jenkins is the way !");
+    JenkinsRule.JSONWebResponse response = j.postJSON("testing-cli/postSomething", jsonBody);
+
+    //because API is returning the same object, we assert the input message.
+    assertTrue(response.getContentAsString().contains("Jenkins is the way !")); 
+    assertEquals(response.getStatusCode(), 200);
+}
+----
+
+== Some additional informations
+
+For people that are familiar with REST/json concept you may want to use other HTTP verbs, it should work, but since generally in Jenkins onlly `GET` and `POST are used this page only shows example for this 2 verbs.
```

Comment:
```suggestion
For people that are familiar with REST/JSON concept you may want to use other HTTP verbs, it should work, but since generally in Jenkins only `GET` and `POST are used this page only shows example for this 2 verbs.
```

(+ typo)